### ISSUE TITLE
Is hflip really random?

### ISSUE THREAD

**Issue by isht7** [tom-roddick/mono-semantic-maps#6]
Thanks very much for releasing the code! 
It seems that you have used horizontal flip as an augmentation during training. But looking at [this](https://github.com/tom-roddick/mono-semantic-maps/blob/bdb6b9b390759d6cf177fccb781195f066412011/src/data/augmentation.py#L17-L27) it seems that the images are not flipped at random but instead they are flipped all the time. Shouldn't it be flipped randomly?

**Comment by pha-nguyen** [comment]
thanks for raising the same concern!

**Comment by unreal4sim** [comment]
Apparently not - see code (`src/data/augmentation.py`)

`class AugmentedMapDataset(Dataset):

    def __init__(self, dataset, hflip=True):
        self.dataset = dataset
        self.hflip = hflip
    
    def __len__(self):
        return len(self.dataset)
    
    def __getitem__(self, index):
        image, calib, labels, mask = self.dataset[index]

        # Apply data augmentation
        if self.hflip:
            image, labels, mask = random_hflip(image, labels, mask)

        return image, calib, labels, mask
 
def random_hflip(image, labels, mask):
    image = torch.flip(image, (-1,))
    labels = torch.flip(labels.int(), (-1,)).bool()
    mask = torch.flip(mask.int(), (-1,)).bool()
    return image, labels, mask`

But I am also discovering that the validation output is 'flipped'. We might have to unflip to get the correct output.

**Comment by unreal4sim** [comment]
That is correct. I am also discovering that the validation data is flipped (i.e. labels and scores). It seems to be necessary to unflip (flip again) to view them correctly.
